### PR TITLE
[MFTF] Refactoring of AdminConfigDefaultProductLayoutFromConfigurationSettingTest

### LINKED ISSUE #30855: [Issue] [MFTF] Refactoring of AdminConfigDefaultProductLayoutFromConfigurationSettingTest
This issue is automatically created based on existing pull request: magento/magento2#30850: [MFTF] Refactoring of AdminConfigDefaultProductLayoutFromConfigurationSettingTest

---------
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Go to Admin->Stores->Configuration->General->Web->Default layouts
2. Set Default Product Layout to "3 column" and sve changes
3. Start creating a new simple product
4. Check if Design->Layout is "3 column"

### PR DESCRIPTION
**Pull request by AnnaAPak**
### Description (*)
The test is refactored according to the best practices followed by MFTF.

### Manual testing scenarios (*)
1. Go to Admin->Stores->Configuration->General->Web->Default layouts
2. Set Default Product Layout to "3 column" and sve changes
3. Start creating a new simple product
4. Check if Design->Layout is "3 column"

### Resolved issues:
1. [x] resolves magento/magento2#30855: [MFTF] Refactoring of AdminConfigDefaultProductLayoutFromConfigurationSettingTest

### CODE REVIEW

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by sidolov** [comment]
@magento create issue

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 1]
```diff
@@ -21,18 +21,38 @@
             <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
         </before>
         <after>
-            <actionGroup ref="RestoreLayoutSetting" stepKey="sampleActionGroup"/>
+            <actionGroup ref="NavigateToDefaultLayoutsSettingActionGroup" stepKey="navigateToWebConfigurationPage1"/>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 2]
```diff
@@ -21,18 +21,38 @@
...
+            <actionGroup ref="AdminSetProductLayoutSettingsActionGroup" stepKey="sampleActionGroup">
+                <argument name="layout" value="1 column"/>
+            </actionGroup>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 3]
```diff
@@ -21,18 +21,38 @@
...
+            <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCacheBeforeTestFinishes">
+                <argument name="tags" value="config"/>
+            </actionGroup>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 4]
```diff
@@ -21,18 +21,38 @@
...
-        <waitForElementVisible selector="{{DefaultLayoutsSection.productLayout}}" stepKey="DefaultProductLayout"/>
-        <selectOption selector="{{DefaultLayoutsSection.productLayout}}" userInput="3 columns" stepKey="select3ColumnsLayout"/>
-        <click selector="{{ContentManagementSection.Save}}" stepKey="clickSaveConfig"/>
-        <amOnPage url="{{AdminProductCreatePage.url(AddToDefaultSet.attributeSetId, 'simple')}}" stepKey="navigateToNewProduct"/>
-        <waitForPageLoad stepKey="wait1"/>
-        <click selector="{{ProductDesignSection.DesignTab}}" stepKey="clickOnDesignTab"/>
-        <waitForElementVisible selector="{{ProductDesignSection.LayoutDropdown}}" stepKey="waitForLayoutDropDown"/>
+
+        <actionGroup ref="NavigateToDefaultLayoutsSettingActionGroup" stepKey="navigateToWebConfigurationPage"/>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 5]
```diff
@@ -21,18 +21,38 @@
...
+        <comment userInput="Comment is added to preserve the step key for backward compatibility" stepKey="expandDefaultLayouts"/>
+        <comment userInput="Comment is added to preserve the step key for backward compatibilityr" stepKey="DefaultProductLayout"/>
+        <actionGroup ref="AdminSetProductLayoutSettingsActionGroup" stepKey="select3ColumnsLayout">
+            <argument name="layout" value="3 columns"/>
+        </actionGroup>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 6]
```diff
@@ -21,18 +21,38 @@
...
+        <actionGroup ref="CliCacheFlushActionGroup" stepKey="clickSaveConfig">
+            <argument name="tags" value="config"/>
+        </actionGroup>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 7]
```diff
@@ -21,18 +21,38 @@
...
+        <actionGroup ref="AdminOpenNewProductFormPageActionGroup" stepKey="navigateToNewProduct"/>
+        <comment userInput="Comment is added to preserve the step key for backward compatibility" stepKey="wait1"/>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 8]
```diff
@@ -21,18 +21,38 @@
...
+        <actionGroup ref="AdminExpandProductDesignSectionActionGroup" stepKey="clickOnDesignTab"/>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml [hunk 9]
```diff
@@ -21,18 +21,38 @@
...
+        <comment userInput="Comment is added to preserve the step key for backward compatibility" stepKey="waitForLayoutDropDown"/>
         <seeOptionIsSelected selector="{{ProductDesignSection.LayoutDropdown}}" userInput="3 columns" stepKey="see3ColumnsSelected"/>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]:
```suggestion
```

File: app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminSetProductLayoutSettingsActionGroup.xml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        <arguments>
+            <argument name="layout" type="string"/>
+        </arguments>
+        
+        <waitForElementVisible selector="{{DefaultLayoutsSection.productLayout}}" stepKey="waittForDefaultProductLayout"/>
+        <selectOption selector="{{DefaultLayoutsSection.productLayout}}" userInput="{{layout}}" stepKey="selectLayout"/>
+        <click selector="{{ContentManagementSection.Save}}" stepKey="clickSaveConfig"/>
+        <waitForPageLoad stepKey="waitForSavingSystemConfiguration"/>
+        <see userInput="You saved the configuration." stepKey="seeSuccessMessage"/>
+
```
Comment (eduard13) [inline app/code/Magento/Cms/Test/Mftf/ActionGroup/AdminSetProductLayoutSettingsActionGroup.xml]:
```suggestion
```

File: app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminExpandProductDesignSectionActionGroup.xml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AdminExpandProductDesignSectionActionGroup">
+        <annotations>
+            <description>Expand the Design section on the Product Details page in Admin.</description>
+        </annotations>
+
+        <click selector="{{ProductDesignSection.DesignTab}}" stepKey="clickDesignTab"/>
+        <waitForPageLoad stepKey="waitForTabOpen"/>
+
```
Comment (eduard13) [inline app/code/Magento/Catalog/Test/Mftf/ActionGroup/AdminExpandProductDesignSectionActionGroup.xml]:
```suggestion
```

**Review by eduard13 — APPROVED** [review]
Looks good. Just for your information, please avoid adding a new row after each action. Could you please apply the code suggestions? Looks like there are no write permissions to your fork.
Thank you.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8541](https://jira.corp.magento.com/browse/ENGCOM-8541) has been created to process this Pull Request
:eight_spoked_asterisk: @eduard13, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by eduard13** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml (hunk 1 above)
Comment (AnnaAPak) [reply app/code/Magento/Catalog/Test/Mftf/Test/AdminCreateSimpleProductTest/AdminConfigDefaultProductLayoutFromConfigurationSettingTest.xml]: @eduard13  thank you, I removed empty rows.

**Comment by AnnaAPak** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by eduard13** [comment]
@magento run all tests

**Review by eduard13 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8541](https://jira.corp.magento.com/browse/ENGCOM-8541) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
Moving this PR to _Extended Testing_ column as this is an MFTF Refactoring PR

**Comment by engcom-Hotel** [comment]
@magento run Functional Tests B2B

**Pull request merged**